src/Service/SessionConfigFactory.php

```diff
@@ -55,6 +64,23 @@ public function __invoke(ContainerInterface $container, $requestedName, ?array $
             unset($config['config_class']);
         }
 
+        if (
+            $container->has(SaveHandlerInterface::class)
+            && (! isset($config['save_handler']) || $config['save_handler'] === SaveHandlerInterface::class)
+        ) {
+            $saveHandler = $container->get(SaveHandlerInterface::class);
```

Comment:
You should have a `has()` check here _first_, so you can throw a more specific error (e.g. "SaveHandlerInterface specified as session save_handler, but the service is not registered in the container").

Reply:
Thanks for the review!

The `->has()` check is on line 68, as if the service is not registered there is no error, it behaves as before to maintain BC and also this is how SessionManager does it.

I could add an additional check if save_handler is explicitly set to SaveHandlerInterface and its not set to throw?

Reply:
updated.